File: backend/src/main/java/com/viniciusfinger/rsflood/dto/AddressDTO.java
package com.viniciusfinger.rsflood.dto;

import com.viniciusfinger.rsflood.enums.HomeZone;
import lombok.Data;

import java.util.List;

@Data
public class AddressDTO {

    private Long id;

    private String street;

    private String number;

    private String cep;

    private String neighbourhood;

    private String city;

    private HomeZone zone;

    private String observation;

    private List<HomeDTO> homes;

}

File: frontend/src/app/model/dto/address-dto.ts
import { HomeZone } from "../enum/home-zone";
import { HomeDTO } from "./home-dto";

export class AddressDTO {
    id: number;
    street: string;
    number: string;
    cep: string;
    neighbourhood: string;
    city: string;
    zone: HomeZone;
    observation: string;
    homes: HomeDTO[];

    constructor(
        id: number,
        street: string,
        number: string,
        cep: string,
        neighbourhood: string,
        city: string,
        zone: HomeZone,
        observation: string,
        homes: HomeDTO[]
    ) {
        this.id = id;
        this.street = street;
        this.number = number;
        this.cep = cep;
        this.neighbourhood = neighbourhood;
        this.city = city;
        this.zone = zone;
        this.observation = observation;
        this.homes = homes;
    }
}
